shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -142,4 +162,27 @@ private void processingFailed(Class<?> entity, Throwable t) {
     private void processingComplete(Class<?> entity) {
         LOGGER.debug("[shard] Successfully processed all entities '{}' to deploy or delete", entity.getSimpleName());
     }
+
+    private Uni<HttpResponse<Buffer>> getAuthenticatedRequest(HttpRequest<Buffer> request, Function<HttpRequest<Buffer>, Uni<HttpResponse<Buffer>>> executor) {
+        if (client.isPresent()) {
+        	return Uni.createFrom().deferred(() -> {
+        		tokenLock.lock(); 
```

Comment:
We need synchronization to be sure we have just one token, but that sync is done inside the uni, so although waiting, the  main thread should be not hold